torch/csrc/autograd/function.cpp

```diff
@@ -15,6 +15,26 @@
 
 namespace torch { namespace autograd {
 
+// The current evaluating node. This is useful to assign the current node as a
+// parent of new nodes created during the evaluation of this node in anomaly
+// mode.
+// The whole `current_evaluating_node` processes and functions are placed in this
```

Comment:
@albanD as you said, the previous test doesn't really check the warnings and apparently the nested warnings did not appear anymore. Moving the `current_evaluating_node` back to the cpp file solves this issue.

Reply:
If you put it in the .h then a "new version" is compiled in each cpp file. And so they are not the same.
You can remove this comment I think :)

Reply:
Removed